lucene/queries/src/test/org/apache/lucene/queries/mlt/TestMoreLikeThis.java

```diff
@@ -427,5 +438,70 @@ public void testMultiFieldShouldReturnPerFieldBooleanQuery() throws Exception {
       analyzer.close();
     }
   }
+
+  public void testCustomFrequecy() throws IOException {
+    // define an analyzer with delimited term frequency, e.g. "foo|2 bar|3"
+    Analyzer analyzer = new Analyzer() {
+
+      @Override
+      protected TokenStreamComponents createComponents(String fieldName) {
+        MockTokenizer tokenizer = new MockTokenizer(MockTokenizer.WHITESPACE, false, 100);
+        MockTokenFilter filt = new MockTokenFilter(tokenizer, MockTokenFilter.EMPTY_STOPSET);
+        return new TokenStreamComponents(tokenizer, foo(filt));
+      }
+
+      TokenStream foo(TokenStream input) {
+        return new TokenFilter(input) {
+
+          @Override
+          public boolean incrementToken() throws IOException {
+            final CharTermAttribute termAtt = addAttribute(CharTermAttribute.class);
+            final TermFrequencyAttribute tfAtt = addAttribute(TermFrequencyAttribute.class);
```

Comment:
Normally `addAttribute` calls must happen at the `TokenFilter` construction time -- I think the test only works because the `AttributeFactory` has already added these attributes.  Maybe move them up above?

Reply:
👍